v1_1/system.ts

```diff
@@ -0,0 +1,167 @@
+        this.data = CryptoV1_1.genCode(this.infoStr());
+    }
+
+    static infoStr(name: string, location: string, room: string, locationType: number): string {
+        return [name, location, room, locationType.toString()].join("::");
+    }
+
+    infoStr(): string {
+        return Location.infoStr(this.name, this.location, this.room, this.locationType);
+    }
+
+    /**
+     * Returns the base64 encoded protobuf-message for the location owner.
+     *
+     * @param baseURL - anything - is ignored and removed afterwards.
+     */
+    getQRtrace(baseURL: string): string {
+        const qrct = new QRCodeTrace({
+            version: 2,
+            trSk: this.data.tr.sk,
+            trR2: this.data.tr.r2,
+            pTrR1: this.data.pTr.r1,
+            pTrR2: this.data.pTr.r2,
+            info: this.infoStr(),
+        });
+        const traceBuf = QRCodeTrace.encode(qrct).finish();
+        const traceEnc = crypto_box_seal(traceBuf, this.healthAuthorityPubKey);
+        return `${baseURL}#${to_base64(traceEnc)}`;
+    }
+
+    /**
+     * Returns the base64 encoded protobuf-message necessary for visitors to register.
+     *
+     * @param baseURL - anything - is ignored and removed afterwards.
+     */
+    getQRentry(baseURL: string): string {
+        const qrCodeContent = QRCodeContent.create({
+            version: 2,
+            publicKey: this.data.ent,
+            pEnt: this.data.pEnt,
+            name: this.name,
+            location: this.location,
+            room: this.room,
+            venueType: this.locationType,
+            notificationKey: this.notificationKey
+        });
+
+        const qrCodeBuf = QRCodeContent.encode(qrCodeContent).finish();
+        return `${baseURL}#${to_base64(qrCodeBuf)}`;
+    }
+}
+
+/**
+ * The user has zero or more instances of Visit in his phone.
+ */
+export class Visit {
+    private visit: IVisit;
+
+    constructor(
+        qrCodeEntry: string,
+        public entry: number,
+        public departure: number,
+        diary?: boolean
+    ) {
+        const qrBase64 = qrCodeEntry.replace(/^.*#/, '');
+        const content = QRCodeContent.decode(from_base64(qrBase64));
+        if (content === undefined) {
+            throw new Error("didn't find 'content'");
+        }
+        if (content.version !== 2) {
+            throw new Error("Wrong version")
+        }
+        const infoBuf = Location.infoStr(content.name, content.location, content.room,
+            content.venueType);
+        this.visit = CryptoV1_1.scan(content.publicKey, content.pEnt, infoBuf, diary);
+    }
+
+    /**
+     * Uses the traces created by the healthAuthority to detect if it has been exposed.
+     *
+     * @param traces sent by the health authority
+     * @return true if at least one of the traces is positive
+     */
+    verifyExposure(traces: ITrace[]): (undefined | string) {
+        for (const trace of traces) {
+            const info = CryptoV1_1.match(this.visit, trace.tr);
```

Comment:
following the convention, this should be `aux` not `info`. Aux should then be parsed to recover the notification key and entry + exit times. (Parsing can also happen inside `match`).

Reply:
fixed